opencog/nlp/microplanning/anaphora-nouns-list.scm

```diff
@@ -99,8 +99,8 @@
 ; 'proc' needs to be a normal Scheme procedure (not <generic>) but must
 ; accept a <noun-item> as the parameter.
 ;
-(define-method (for-each-by-proc (nl <nouns-list>) (proc <procedure>))
-	(for-each proc (slot-ref nl 'lst))
+(define-method (par-for-each-by-proc (nl <nouns-list>) (proc <procedure>))
+	(par-for-each proc (slot-ref nl 'lst))
 )
```

Comment:
FWIW, parallelism will work best if its done at the highest level, not a low level.    For example, process 6 sentences in parallel, instead of one sentence in 6 threads ...